shinyswatch/_theme_utils.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+from htmltools import HTMLDependency, Tagifiable, TagList
+
+from ._assert import assert_theme
+from ._bsw5 import BSW5_THEME_NAME, bsw5_theme_colors
+from ._get_theme_deps import get_theme_deps as _get_theme_deps
+
+
+class ThemeColors:
+    def __init__(self, name: BSW5_THEME_NAME) -> None:
+        self._name = name
+        self._colors: dict[str, str] = bsw5_theme_colors[name]
+
+    def __getattr__(self, key: str) -> str:
+        return self._colors[key]
+
+    def __getitem__(self, key: str) -> str:
+        return self._colors[key]
+
+    def __dir__(self) -> list[str]:
+        return list(self._colors.keys())
+
+    def __repr__(self) -> str:
+        colors = [f"{k:11}: {v}" for k, v in self._colors.items()]
+        colors = "\n ".join(colors)
+        ret = [f"<ThemeColors({self._name!r}):", " " + colors, ">"]
+        return "\n".join(ret)
+
+
+class ShinyswatchTheme(Tagifiable):
+    def __init__(self, name: BSW5_THEME_NAME) -> None:
+        assert_theme(name=name)
+        self.name: BSW5_THEME_NAME = name
+        self.colors = ThemeColors(name)
+
+    def __call__(self) -> list[HTMLDependency]:
+        return _get_theme_deps(self.name)
+
+    def tagify(self) -> TagList:
```

Comment:
This is a compromise to make `ShinyswatchTheme` inherit `Tagifiable`. Calling the class returns `list[HTMLDependency]`, but the `.tagify()` method returns a `TagList`.

The benefit is that you can use the functional syntax in `ui.page_navbar()`.

```python
# throws
ui.page_navbar(
	shinyswatch.theme.minty,
	ui.nav_panel()
)

# works
ui.page_navbar(
	shinyswatch.theme.minty(),
	ui.nav_panel()
)
```

Reply:
I tried an alternate version where `get_theme_deps()` returns the `TagList`, but we ended up needing to narrow the types down in other places (like `theme_picker_server()`) with code like this:

```python
[dep for dep in list(get_theme_deps("minty")) if isinstance(dep, HTMLDependency)]
```

In the end it was just a lot easier everywhere to have the `list[HTMLDependency]` that's upgraded to a `TagList` where needed.